src/SymfonyCache/CustomTtlListener.php

```diff
@@ -59,15 +59,16 @@ public function __construct($ttlHeader = 'X-Reverse-Proxy-TTL', $keepTtlHeader =
     public function useCustomTtl(CacheEvent $e)
     {
         $response = $e->getResponse();
-        if (!$response->headers->has($this->ttlHeader)) {
-            return;
-        }
```

Comment:
This currently behaves that `s-maxage` wins when a CustomTtlListener is enabled but header not defined. This I think is not the expected behaviour of the CustomttlListener and leads so to unexpected responses being cached (in our case big media files).

Reply:
that is by design. if s-maxage is set, any random proxies between the client and your server might cache the response. why would you specifically not want to cache in your reverse proxy, but in uncontrolled proxies somewhere else?

you mention large media files you don't want to cache: for that use case, i would either write your own listener that checks the size and disables caching for the response, or have the application send a custom ttl of 0. i would recommend the custom listener, as it would logically be okay to cache the response, but from a practical point of view not suitable. the drawback would be that you have no control of telling in the cache layer whether the response was expensive to generate or not. (btw, varnish cache allows to have 2 separate cache stores for small and large responses to avoid large responses bombing out bunches of small ones.). and another btw: if you deliver large media files from the filesystem (e.g. for access control) you could look into x-accel / x-sendfile to make your webserver do the heavy lifting and avoid piping binary data through php. that would also have the side effect of being able to cache the response again, as the symfony cache happens before the web server resolves x-accel.

Reply:
> Why would you specifically not want to cache in your reverse proxy, but in uncontrolled proxies somewhere else?

Because the s-maxage is in our case for third party proxies but not for our own caches. Think https://github.com/symfony/symfony/issues/47288 would still be best solution for such cases to control what layer caches what things.

The media are not files in this case and so we can not work with the BinaryFileResponse of Symfony in this case / x-sendfile.

---

What do you think about an additional flag for the `CustomTtlListener`? See may latest changes.